src/foam/u2/detail/SectionedDetailPropertyView.js

```diff
@@ -262,28 +262,36 @@ foam.CLASS({
 
   requires: [
     'foam.u2.layout.Cols',
-    'foam.u2.layout.Rows'
+    'foam.u2.layout.Rows',
+    'foam.u2.Visibility'
   ],
 
   properties: [
     'prop',
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.Slot',
+      name: 'visibilitySlot',
```

Comment:
Hmm thinking about this a little more right now. Would it make life simpler if we changed this to:
```
{
  class: 'Enum',
  of: 'foam.u2.Visibility',
  name: 'visibility'
}
```
and in `initE` we do:
```
this.onDetach(this.visibility$.follow(this.ProxySlot.create({
  delegate$: this.prop$.map(p => {
    p.createVisibilityFor(this.data$).map((m) => m !== this.Visibility.HIDDEN)
  }
// brackets...
```

Maybe I'm overthinking this lol

Reply:
The reason we needed to move the visibility slot into a property is so we could reference it from `SectionView`. I think if we made the change you're suggesting here we'd only be able to access the visibility, but not the visibility slot. Of course we could duplicate the logic but that's what I did initially and we changed it so that wasn't necessary.

Reply:
Yeah but I think the way you have it is fine. I went too far lol